### PR TITLE
Correctly set default options for boolean flags

### PR DESCRIPTION
**Pull request by mrm007**
See #71.

This is a follow-up to #63 where I removed the default values for some boolean flags (`semi`, `use-tabs`, `bracket-spacing`, `jsx-bracket-same-line`, `single-quote`).

yargs still sets the default value to `false` if the option is not present. This PR sets the default value to `undefined` as per [the documentation](https://github.com/yargs/yargs#booleankey).

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -112,10 +114,10 @@ const parser = yargs
         - true - Example: { foo: bar }
         - false - Example: {foo: bar}
       `,
-      choices: [true, false],
```
Comment (kentcdodds) [inline src/parser.js]: Why this change?

Comment (mrm007) [inline src/parser.js]: I didn't make much sense when I added it in the previous commit. It's encouraged to be used as either `--semi` or `--no-semi`. Furthermore, yargs complains that `undefined` is not a valid choice (`true` or `false`).

Comment (kentcdodds) [inline src/parser.js]: Ok, cool! 👍

**Comment by kentcdodds** [comment]
Thank you!
